### ISSUE TITLE
Official Release ETA

### ISSUE THREAD

**Issue by jpike88** [sequelpro/sequelpro#3385]
I've been riding on nightly builds for months, but the latest official release still has some pretty huge bugs on Mojave.

I think it's a good time to get the current nightly out there, before too much of the user base is using a nightly build for their daily driver.

**Comment by itmayziii** [comment]
Yep I'm so tired of this app crashing any time I close a tab.

**Comment by takizo** [comment]
> I've been riding on nightly builds for months, but the latest official release still has some pretty huge bugs on Mojave.
> 
> I think it's a good time to get the current nightly out there, before too much of the user base is using a nightly build for their daily driver.

+1 on this :) I am also using the latest build, it would be nice if there is official release. Since version 1.2 and Version 2 is in the pipeline. 

Thanks for the hard work from fellow contributors.

**Comment by ccharlton** [comment]
+1

**Comment by JohnYeohMY** [comment]
Following a popular PHP framework's author, I too have switched to use TablePlus.  
While it is not free, it is a good alternative to sequelpro.

**Comment by happyendin** [comment]
that's correct, before Mojave is very stable.

**Comment by tinusg** [comment]
> Following a popular PHP framework's author, I too have switched to use TablePlus.
> While it is not free, it is a good alternative to sequelpro.

I prefer Sequel Pro's interface and even if Sequel Pro wasn't "free" (I donated), I'd buy it just to support it. If you're a developer and use your DB GUI on a daily basis and you can afford it, it shouldn't matter if it's free or not.

**Comment by JohnYeohMY** [comment]
> > Following a popular PHP framework's author, I too have switched to use TablePlus.
> > While it is not free, it is a good alternative to sequelpro.
> 
> I prefer Sequel Pro's interface and even if Sequel Pro wasn't "free" (I donated), I'd buy it just to support it. If you're a developer and use your DB GUI on a daily basis and you can afford it, it shouldn't matter if it's free or not.

The problem is that people are not donating enough to allow project owner to dedicate enough time for bug fixes and releasing the updates. Yet these people still comes here and complain about it.

You all should be ashamed of yourselves for bugging someone who is dedicating his time to this free software. Asking for so many things yet offering nothing in return.

**Comment by stychos** [comment]
And you think that pr'ing of another paid product is better in this situation? Lol.

**Comment by JohnYeohMY** [comment]
> And you think that pr'ing of another paid product is better in this situation? Lol.

Thanks for showing everyone how ignorant you are.

TablePlus is free for use with limitations. You can only open up to 2 tabs but at least the software doesn't crash when you close a tab.

**Comment by stychos** [comment]
Thanks for showing how ignorant you are, thinking that everyone must scaff to install every application mentioned by every user and immediately lurking one's capabilities from top to bottom. Or it isn't you who wrote following?
> While it is not free

**Comment by JohnYeohMY** [comment]
> Thanks for showing how ignorant you are, thinking that everyone must scaff to install every application mentioned by every user and immediately lurking one's capabilities from top to bottom. Or it isn't you who wrote following?
> 
> > While it is not free
> 
> Would like to help here? Make a donation to authors, contribute some code. Don't advertise concurrent applications, those who search will find.

And yet again you are showing your ignorance. Check the donations page under October 2018. My name is there.

I'm only suggesting alternatives to those whiners who didn't contribute a thing yet demand so much. If you don't have anything nice to say, keep your mouth shut and get off your high horse.

**Comment by Kaspik** [comment]
Chill guys.

Still valid what we said - bugs in https://github.com/sequelpro/sequelpro/projects/2 need to be fixed for 1.2 release, but we are moving there. Nightly builds are stable, if anyone is able to help with PRs to get these bugs fixed, even better. We are fully ready to release new version once all of them are fixed! :)

**Comment by vdhub** [comment]
> I've been riding on nightly builds for months, but the latest official release still has some pretty huge bugs on Mojave.
> 
> I think it's a good time to get the current nightly out there, before too much of the user base is using a nightly build for their daily driver.

I guess they should , the tool is nice and effective but the crash every time on Mojave is a nightmare.

**Comment by jpike88** [comment]
@Kaspik there's a pretty critical bug fix regarding the crash on tab closing on Mojave, shouldn't a 1.19 version be released asap to correct that? I'm on nightly builds now but on the official release it was driving me insane, it's a showstopper for people who are using Mojave and it's been out for over 6 months now

**Comment by schmod** [comment]
Everybody in this thread is aware of the tab bug.

**Comment by jpike88** [comment]
@schmod Did I imply anyone wasn't? What's your point?

**Comment by beeradmoore** [comment]
@jpike88 Mojave may have been out for over 6 months but this product hasn't had an official stable update released in its last 220 commits over the past 3 years. I only just found that nightly builds exist :o

**Comment by cakoose** [comment]
Does anyone know the commit/commits that fixes the Mojave tab crash issue?  I'm hoping to cherry-pick onto 1.1.2 and make a build for myself.

**Comment by beeradmoore** [comment]
@cakoose , possibly any of these? https://github.com/sequelpro/sequelpro/search?q=tab+crash&type=Commits

Although that is pre-mojave release.

**Comment by cakoose** [comment]
Good idea, @beeradmoore.  Unfortunately, those cherry-picks aren't applying cleanly :-\  I guess I'll just wait for the next release...

**Comment by andy-laravel** [comment]
> Would like to help here? Make a donation to authors, contribute some code. 

While I'd love to have contributed some way, me and by the looks of it, quite a lot of people see that nothing has been done to address the *critical/urgent* issue of crashing on closing tab, so it would be hard to justify making any donation because it kind of seems we're being ignored. Why is it taking 9 months or so to fix one *major* issue?

**Comment by anotherdevs** [comment]
@andy-laravel The bug has been fixed in the nighlies: https://sequelpro.com/test-builds however @abhibeckert and @mattlangtree has left the project with the keys to sign the program, so any official release is not possible. The right thing is to stop donating as the money is not spent on this project.

**Comment by jpike88** [comment]
@Kaspik I'd appreciate an answer... I find it really weird that certain people are ok with how broken the offical release is on Mojave, when it takes barely any effort to get a release out there for the users struggling on Mojave (which is probably a large portion of userbase now). I'm struggling to understand how that isn't a bigger priority.

**Comment by jpike88** [comment]
If the keys are so hard to get a hold of, maybe the solution here to is fork it under a new name.

**Comment by takizo** [comment]
> If the keys are so hard to get a hold of, maybe the solution here to is fork it under a new name.

I support on this. I would love to see this project keep on going.

**Comment by JohnYeohMY** [comment]
> @abhibeckert and @mattlangtree has left the project with the keys to sign the program, so any official release is not possible. The right thing is to stop donating as the money is not spent on this project.

Where did you hear this from? I feel so betrayed now after making donations and defending them.

**Comment by andy-laravel** [comment]
> I would urge anyone to abandon ship and use TablePlus: https://tableplus.io/. You can sign up with SetApp: https://setapp.com/

But its not free/opensource so it's not really comparable to Sequel Pro

**Comment by JohnYeohMY** [comment]
> > I would urge anyone to abandon ship and use TablePlus: https://tableplus.io/. You can sign up with SetApp: https://setapp.com/
> 
> But its not free/opensource so it's not really comparable to Sequel Pro

Sometimes, it's better to pay for a product so that the developers can afford to continue to maintain it. Without a steady stream of income, you end up with another tragic SequelPro story.

**Comment by andy-laravel** [comment]
> @andy-laravel The bug has been fixed in the nighlies: https://sequelpro.com/test-builds 
But they introduce new issues like the weird filtering - and unable to remove filter issue 
We just want a simple fix to the original issue. 
No new features, just a fix. Is it really that hard?

**Comment by jpike88** [comment]
I used the trial of Querious for a bit, does anyone know if it's decent, and what its like compared to TablePlus?

**Comment by Kaspik** [comment]
@anotherdevs Not true, please stop sharing lies.
@jpike88 We have keys and everything, we just don§t have time to fully focus on the last bugs that are mentioned as "required" for release.
@JohnYeohMY Your donations are highly appreciated and thanks to them the nightly builds and builds servers are running! :)
@cakoose Yeah, we tried, but there is too much stuff around that cherry pick is not possible.

Nightly build works. Is stable, I'm using it daily. We all have jobs and personal life, so if you want / can, contribute not by donations, but by PR for the rest of the bugs in https://github.com/sequelpro/sequelpro/projects/2

If there is nothing related to this bug, I would like to ask you to stop the conversation OOT, thank you.

**Comment by jpike88** [comment]
> last bugs that are mentioned as "required" for release

Seems pretty arbitrary to me, official release seems a bit desperately overdue at this point IMO. Best it's good to at least see things are still alive

**Comment by anotherdevs** [comment]
@JohnYeohMY You can search this forum: 
https://github.com/sequelpro/sequelpro/issues/3377#issuecomment-461317102
https://github.com/sequelpro/sequelpro/issues/3034#issuecomment-424656667
https://github.com/sequelpro/sequelpro/pull/3192#issuecomment-426269204
https://github.com/sequelpro/sequelpro/issues/3074#issuecomment-395879044
@Kaspik Do you have the keys?

**Comment by Kaspik** [comment]
Yes, we do have them as well as all the other access credentials to all services.

**Comment by anotherdevs** [comment]
@Kaspik However @stychos just one month ago stated otherwise. You are not part of the core team, so your word is as unreliable as @stychos. The fact however is that @dmoagx has stated he will fork this project and leave it. He also stated that HE didn't have the keys and ability to launch a new release. I can see that you don't like it, which is sad.
I do still get notifications on all these issues as I have subscribed to them and no (good) news since last year around november I think. :-)

**Comment by stychos** [comment]
Hm.. @anotherdevs, I am not related to the Sequel Pro team in any way, my badge here is added rather accidentally (and later).
That referenced comment seems to be out of context, I thought that the project is abandoned, but the whole discussion and periodic team actions showed that it isn't fully true. My personal opinion is that the project needs community's help: time is a very critical resource for everyone.

**Comment by jpike88** [comment]
So much nonsense could be averted if a release was performed now, and these other bugs be fixed when people have the time. @Kaspik I still don't get why you'd hold back a release all this time, for the sake of some bugs that nobody has a timeline on fixing. It's not hard to perform a release, and this is the third time I've asked you this. A direct answer would be appreciated. And if time is your reason for being unable to perform a release, I'll do it for you.

**Comment by andy-laravel** [comment]
I echo the comments above, if it is only time that is the issue, pass it on to the community to release. It needs to be done before you core fanbase fades away. You must admit its been too long since the last official release and there needs to be action to fix a critical bug

**Comment by ccharlton** [comment]
Hi, everyone. We need working software. That's all.

Thanks, bye.

**Comment by schmod** [comment]
Can we please close this thread?

* The maintainers have emphasized that they are still here.
* The maintainers have clearly stated which bugs they believe are blockers for the release
* These bugs are still open
* Nightly builds are available that fix the tab issue, but still have the other bugs present.  Many users have reported success with these builds.
* Numerous threats to fork the project have been made

If you would like to fork the project, go ahead and do so.  If you would like to fix the remaining bugs, you're free to do that too.  Complaining at and harassing the (mostly-volunteer, part-time) maintainers is not productive, and has already been done _ad nauseam_.

**Comment by dmoagx** [comment]
> @JohnYeohMY Your donations are highly appreciated and thanks to them the nightly builds and build servers are running! :)

I have to say, I personally disagree with this statement (at least at this point in time).

**Comment by jpike88** [comment]
> The maintainers have clearly stated which bugs they believe are blockers for the release

If questioning the rationale of delaying a release with the current state of the official version, which multiple times now has not been directly answered, then offering to perform a release which is minimal effort and can only help the poor users out there who aren't riding the nightly build equates to harassing the maintainers, then that's unfortunate.

But sure I'll close it, because this isn't going anywhere anyway.

**Comment by beeradmoore** [comment]
I wouldn't want to use nightly builds to connect to my DBs. Imagine if they had a bug that could accidentally delete the wrong table. 

Jumping back to MySQL Workbench. At the moment not sure why I didn't do this from the start.

**Comment by origamifreak2** [comment]
I don't know if this idea would actually help or how useful it would actually be, but is there a place where users could report which version of the nightly build they're using, how they're using it, and whether or not they've had any major issues? This could allow someone to more easily find a nightly build that works for what they need without being nervous that there's a bug somewhere that could harm their data.

For example, I've been using build d78907bd2b with MySql 5.6.40 and 10.1.22-MariaDB for a few months now without any noticeable data issues.

Also, a post on https://sequelpro.com/news might help people know that development is still ongoing.

**Comment by jpike88** [comment]
Last commit and nightly build was 2 months ago, something tells me that expecting anything more at this point is futile.

@schmod "The maintainers have emphasized that they are still here" Define 'here'. You mean, alive with a GitHub account?

**Comment by stychos** [comment]
@jpike88 agree with you, I feel here is no maintainers and the project is dead. Only one single developer periodically doing something **he** personally need, also, rejecting most of side work because of his needs to keep maintainability with the OS X 10.6 (hell, who the f_ck tied him to this old crap? — even browsers didn't work with it for a long time). But generally, here is no maintainer who will review pull requests, give a direction for volunteers or pass that privileges to someone else.

**Comment by jpike88** [comment]
Oh yeah, #3116, STILL not on main release. When Mojave came out over 6 months ago.

It's probably in my whole career the most dysfunctional OSS project I've ever seen. A good lesson in how to not run a project.

**Comment by rowanbeentje** [comment]
Right now I don't think there's anyone running the project - the developer who has contributed the last commits does care about supporting old systems *and* releasing a stable version, but doesn't agree with the current project ownership and governance so doesn't want to commit to releasing the current version under the current name. As a past contributor with no current involvement I don't feel comfortable with going against his wishes.

"Main release" is the exact same code and build system as the nightlies, so for now you're not losing anything by using the nightlies - it won't magically become more stable by being tagged as main.  If you're arguing for a release now, it would be the same as the nightly so do feel free to use that.

**Comment by jpike88** [comment]
If a fork is released, I'm happy to help organise it... many services are free for an OSS project

**Comment by JohnnyWalkerDesign** [comment]
There's only one thing worse than having software that people are complaining about: Having software nobody cares about. I hope the devs appreciate the lucky position they're in.

**Comment by jpike88** [comment]
I moved to TablePlus a while ago. It’s free usage should be adequate for most users.

**Comment by dnl-jst** [comment]
Did that too. Bought a license for it and there is also a small chance there will be a Linux version of TablePlus. 👍

**Comment by fuzzy76** [comment]
TLDR; The current release crashes easily, the nightly is more stable than the current release, and cosmetic fixes are deemed important enough to block fixing the unstable release.

For anyone looking for something free (but Java-dependant, I know...), DbVisualizer is ok.

**Comment by vinilanzoni** [comment]
I also moved to TablePlus. And in addition of MySQL, they support others engines such PostgreSQL, SQL Server...

**Comment by Hubertnator** [comment]
It's very sad that this software has no support... :-(

**Comment by tothsanka** [comment]
Sorry to say that, but after years of Sequel Pro fan, I switched to https://www.araelium.com/querious It is amazing really, and not that much expensive.

**Comment by S8ER01Z** [comment]
Is this officially dead now? Asking for a friend.

**Comment by jpike88** [comment]
It’s dead.

**Comment by Jason-Morcos** [comment]
Forked Sequel Pro into "Sequel Ace"
Still would have a lot of work to do, but reserved the name on macOS App Store and would love to work with some other dedicated folks to release a new, still totally free, "Sequel" tool. 
Dropped support for macOS < 10.13 and it doesn't currently compile (due to errors in SPSQLParser). It would need lots of work, but I'd gladly accept any and all PRs and I would be excited to add collaborators to the project.

https://github.com/Sequel-Ace/Sequel-Ace/tree/2.0-Dev

**Comment by wirdehall** [comment]
> https://github.com/Sequel-Ace/Sequel-Ace/tree/2.0-Dev

That link is dead but the project is very much alive it seems. I just downloaded it and it seems awesome!
Checkout this link instead: https://github.com/Sequel-Ace/Sequel-Ace
